workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/metrics.py

```diff
@@ -67,38 +65,23 @@ def _mask_array(
     return masked_arr
 
 
-@curry
-def add_to_metrics(metricname: str, func: Callable[[xr.Dataset], xr.Dataset]):
-    """Register a function to be used for computing metrics
-
-    This function will be passed the diagnostics xarray dataset,
```

Comment:
The refactor to have a common `Registry` class is nice and streamlined, though a side effect is that the information formerly in the `add_to_x` docstrings about what each registry expects and outputs is gone. Please update the documentation section "Developing this workflow" to include this information.

Reply:
Good point. Maybe each registry's merge function is a good place for this docstring, since that function is what distinguishes each registry. Will also update the docs.

Reply:
In the end I just added some comments next to the registry definition. Maybe the right way to do this is to require the developer to specify the function signatures when they initialize the registry but I'm not sure if it's worth the complexity to implement.